x/mock/simulation/random_simulate_blocks.go

```diff
@@ -217,8 +220,10 @@ func createBlockSimulator(testingMode bool, tb testing.TB, t *testing.T, event f
 		// shouldn't happen
 		return ops[0].Op
 	}
-	return func(blocksize int, r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context,
+
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context,
 		accounts []Account, header abci.Header, logWriter func(string)) (opCount int) {
+		lastBlocksizeState, blocksize = getBlockSize(r, lastBlocksizeState, avgBlockSize)
```

Comment:
Where is lastBlocksizeState used apart from the initial function call?

Reply:
It's used in every call to get block size? It's a Markov chain, so it needs the last state.

Reply:
Ok, walked through it and it makes sense to me now. Although I still think it can still be somewhat confusing to anyone looking at the code for the first time because the variables are defined outside the scope of the function definition. 👍